code/game/machinery/martian_printer.dm

```diff
@@ -9,11 +9,17 @@
 	idle_power_usage = 20
 	active_power_usage = 5000
 	var/building = 0
+	var/cooldown_duration = 6000 // 10 minutes
+	var/cooldown_time = 0
+	var/cooldown_state = 0
 	var/print_path = /mob/living/carbon/complex/martian
 
 /obj/machinery/mob_printer/attack_ghost(var/mob/dead/observer/O)
 	if(!canSpawn())
 		return
+	if(cooldown_state)
```

Comment:
You could just use coldown_time here too, as long you initial(cooldown_time) to reset it after the cooldown is over. Same deal with the ping sound.